src/Files.App/Filesystem/StorageFileHelpers/FilesystemResult.cs

```diff
@@ -1,29 +1,36 @@
 // Copyright (c) 2023 Files Community
 // Licensed under the MIT License. See the LICENSE.
 
-using Files.Shared.Enums;
-
 namespace Files.App.Filesystem
 {
 	public class FilesystemResult
 	{
 		public FileSystemStatusCode ErrorCode { get; }
 
-		public FilesystemResult(FileSystemStatusCode errorCode) => ErrorCode = errorCode;
+		public FilesystemResult(FileSystemStatusCode errorCode)
+			=> ErrorCode = errorCode;
+
+		public static implicit operator FileSystemStatusCode(FilesystemResult res)
+			=> res.ErrorCode;
+
+		public static implicit operator FilesystemResult(FileSystemStatusCode res)
+			=> new(res);
 
-		public static implicit operator FileSystemStatusCode(FilesystemResult res) => res.ErrorCode;
-		public static implicit operator FilesystemResult(FileSystemStatusCode res) => new(res);
+		public static implicit operator bool(FilesystemResult res)
+			=> res is not null && res.ErrorCode is FileSystemStatusCode.Success;
 
-		public static implicit operator bool(FilesystemResult res) => res is not null && res.ErrorCode is FileSystemStatusCode.Success;
-		public static explicit operator FilesystemResult(bool res) => new(res ? FileSystemStatusCode.Success : FileSystemStatusCode.Generic);
+		public static explicit operator FilesystemResult(bool res)
+			=> new(res ? FileSystemStatusCode.Success : FileSystemStatusCode.Generic);
 	}
 
 	public class FilesystemResult<T> : FilesystemResult
 	{
 		public T Result { get; }
```

Comment:
I think you might move this to its own file

Reply:
What name would we prefer? We would create a new file with 'FilesystemResult', we cannot, however, because there's already.

Reply:
> I started stage 2 first, renaming class name, namespace, and creating own file is stage 1 and so I will do that in the further PR.

Ok, that's fine. I think we could rename it to something like `TypedFilesystemResult`